src/fidasim.f90

```diff
@@ -7146,6 +7182,45 @@ subroutine track(rin, vin, tracks, ntrack, los_intersect)
             tracks(cc+1)%ind = ind
             cc = cc + 2
             ncross = ncross + 1
+        elseif(cell_split.eq.0) then
+            tracks(cc)%pos = ri + 0.5*dT*vn
+            tracks(cc)%time = dT
+            tracks(cc)%ind = ind
+            cc = cc + 1
+        elseif(cell_split.eq.1) then
+            dt2 = 0.0
+            inv_N = 1.0/n_cells
+            split_loop1: do jj=1,int(n_cells)
+                dt2 = dt2 + dT*inv_N
+                tracks(cc)%pos = ri + 0.5*dt2*vn
+                tracks(cc)%time = dT*inv_N
+                tracks(cc)%ind = ind
+                cc = cc + 1
+            enddo split_loop1
+        elseif(cell_split.eq.2) then
+            dt2 = 0.0
+            call get_plasma(plasma1, ri)
+            call get_plasma(plasma2, ri_tmp)
+            dene_sum = plasma2%dene + plasma1%dene
+            if(dene_sum.le.0.0) then
+                inv_ne = 1.0
+            else
+                inv_ne = 2.0/dene_sum
+            endif
+            n_cells = ceiling(abs(plasma1%dene - plasma2%dene)*inv_ne*inv_tol)
+            if(n_cells.gt.50.0) then
+                n_cells = 50.0
+            elseif(n_cells.lt.1.0) then
+                n_cells = 1.0
+            endif
+            inv_N = 1.0/n_cells
+            split_loop2: do k=1,int(n_cells)
+                dt2 = dt2 + dT*inv_N
+                tracks(cc)%pos = ri + 0.5*dt2*vn
+                tracks(cc)%time = dT*inv_N
+                tracks(cc)%ind = ind
+                cc = cc + 1
+            enddo split_loop2
```

Comment:
Add more comments explaining the code. In particular, explain the logic and physics you are implementing. Does this code generalize to other parameters? For example, `dene` looks hardcoded to me. What if the user wants to split according to gradients in the fast-ion density?